lb_film_and_user_recommender.py

```diff
@@ -0,0 +1,251 @@
+    :param num_of_recs: (int) the max number of recommendations to return per returned list
+    :return: (tuple) tuple containing three lists of tuples where list[n][0] = unseen film and list[n][1] = number of
+    users who rated it highly sorted by the number of users. first list is all recommended films, second is more obscure
+    films only, and third is most obscure, calculated by number of ratings
+    """
+
+    all_found_films = {}
+    recommended_films = []
+    obscure_recommended_films = []
+    really_obscure_recommended_films = []
+
+    for result in similar_users:
+        username = re.findall('https://letterboxd.com/(.+)', result[0])[0]
+        found_five_stars = get_users_scores(username, '5')
+        for film in found_five_stars:
+            all_found_films[film] = all_found_films.get(film, 0) + 2
+        found_fourpointfive_stars = get_users_scores(username, '4.5')
+        for film in found_fourpointfive_stars:
+            all_found_films[film] = all_found_films.get(film, 0) + 1.5
+        found_four_stars = get_users_scores(username, '4')
+        for film in found_four_stars:
+            all_found_films[film] = all_found_films.get(film, 0) + 1
+
+    all_found_films = sorted(all_found_films.items(), key=lambda x: x[1], reverse=True)
+
+    for film in all_found_films:
+        activity_str = webpage_to_string(f'https://letterboxd.com/{my_profile}/film/{film[0]}/activity/')
+        if not re.findall('"activity-summary"', activity_str):
+            if len(recommended_films) < num_of_recs:
+                recommended_films.append(f'https://letterboxd.com/film/{films[0]} : {film[1]}')
+            main_page = webpage_to_string(f'https://letterboxd.com/film/{film[0]}/')
+            ratingcount = re.findall('"ratingCount":(.+?),"worstRating"', main_page)
+            if ratingcount:
+                ratingcount = int(ratingcount[0])
+            else:
+                ratingcount = 0
+            if (1000 <= ratingcount < 10000) and (len(obscure_recommended_films) < num_of_recs):
+                obscure_recommended_films.append(f'https://letterboxd.com/film/{films[0]} : {film[1]}')
+            if (ratingcount < 1000) and (len(really_obscure_recommended_films) < num_of_recs):
+                really_obscure_recommended_films.append(f'https://letterboxd.com/film/{films[0]} : {film[1]}')
+        if (len(recommended_films) and len(obscure_recommended_films) and len(
+                really_obscure_recommended_films)) == num_of_recs:
+            break
+    return recommended_films, obscure_recommended_films, really_obscure_recommended_films
+
+########################################################################################################################
+
+myprofile = 'tsar'
+
+favourite_films = {
+    '5': get_users_scores(myprofile, '5'),
+    '4.5': get_users_scores(myprofile, '4.5'),
+    '4': get_users_scores(myprofile, '4')
+}
```

Comment:
Not sure whether you prefer it like this or not, but you can also make less repetitive:
`
favourite_films = {rating: get_users_scores(myprofile, rating) for rating in ['5', '4.5', '4']}
`